tests/utils_test_scaling.py

```diff
@@ -0,0 +1,112 @@
+import logging
+import math
+import re
+
+from rtctools._internal.debug_check_helpers import DebugLevel
+
+
+def create_log_list_scaling():
+    """
+    Creation of a list for the logs which will be filled with the logging information of the
+    problem.
+    :return:
+    """
+
+    class LogRecordsListHandler(logging.Handler):
+        """
+        Handler class to store log entries in a list
+        """
+
+        def __init__(self, records_list):
+            self.records_list = records_list
+            super().__init__()
+
+        def emit(self, record):
+            self.records_list.append(record)
+
+    logs_list = []
+
+    # logger = logging.getLogger("WarmingUP-MPC")
+    logger = logging.getLogger("rtctools")
+    logger.setLevel(logging.INFO)
+
+    logger.addHandler(LogRecordsListHandler(logs_list))
+
+    return logger, logs_list
+
+
+def create_scaling_problem_check(problem_class):
+    """
+    The problem optimization class that needs to be executed is updated with a debuglevel to
+    ensure that the information about the scaling is logged. Furthermore, the logging list is
+    created to allow for backtracking and check scaling information.
+    :param problem_class: The problem optimization class that should be executed
+    :return:
+    The problem optimization class including its debuglevel and the logging list and logger.
+    """
+    # TODO: currently only the default settings can be used to check the scaling.
+    logger, logs_list = create_log_list_scaling()
+
+    class ProblemClassScaling(problem_class):
+        _debug_check_level = DebugLevel.VERYHIGH
+
+    return ProblemClassScaling, logger, logs_list
+
+
+def check_order(dict_values, order_diff):
+    """
+    Checks the difference in order between the lower and upperbound of several problem settings;
+    objective, matrix and rhs.
+    :param dict_values: dictionary with the different problem settings and their respective lower
+    and upperbound
+    :param order_diff: the maximum difference that is allowed.
+    :return:
+    """
+    msg_order = {}
+    for key, value in dict_values.items():
+        if value[0] != 0.0:
+            order = value[1] / value[0]
+        else:
+            order = value[1]
+        if order > order_diff:
+            order_wrong = math.floor(math.log(order, 10))
+            msg_order[key] = (
+                f"The scaling of this problem is not great, for the {key}, the values are {value} "
+                f"which is of the order {order_wrong}, {order_diff} is the maximum allowed order "
+                f"difference"
+            )
+    assert len(msg_order) == 0, msg_order
+
+
+def check_scaling_problem(logs_list, logger, order_diff=1e6):

```

Comment:
input variable "logger" is not listed as a input parameter in the docstring

Reply:
updated